src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
```

Comment:
> Yeah, I suppose with the QoSFilter-style approach, we could prioritize those ahead of the other agent-based ones to make it much less likely that those would be dropped 

The problem here is that prioritization alone doesn't solve the issue of the thundering herd. In order to break a herd up, we need to tell agents to apply a randomized backoff so that their runtimes de-sync. Suspending requests but otherwise processing them as fast as possible may maintain quality of service for routes like `/status` an `/environment_classes` but doesn't seem like it will be able to break a herd up.